frontend/packages/core/src/Feedback/toast.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { Snackbar } from "@material-ui/core";
+
+import type { AlertProps } from "./alert";
+import { Alert } from "./alert";
+
+export interface ToastProps extends AlertProps {
+  duration?: number;
+  onClose?: () => void;
+}
+
+const Toast: React.FC<ToastProps> = ({
+  onClose,
+  severity = "warning",
+  duration = 6000,
+  ...props
+}) => {
+  const [open, setOpen] = React.useState(true);
+
+  const onDismiss = () => {
+    if (onClose !== undefined) {
+      onClose();
+    }
+    setOpen(false);
+  };
+
+  return (
+    <Snackbar
+      open={open}
+      autoHideDuration={duration}
+      onExiting={onDismiss}
```

Comment:
if someone closes the alert will `onDismiss` be triggered twice?

Reply:
Ooh great catch, yeah it will. I'll add a check to see if it's open before triggering.

Reply:
Added an `open` check in `onDismiss`